src/flamenco/capture/solcap.md

```diff
@@ -0,0 +1,256 @@
+
+All PCapNG blocks end with a redundant 4-byte length field for backward navigation. This is not a separate structure, but simply a `uint32` value written at the end of each block
+
+### Buffer Message Internal Header
+
+```c
+struct fd_solcap_buf_msg {
+   uint16_t sig;       /* Message signature/type */
+   uint64_t slot;      /* Solana slot number */
+   uint64_t txn_idx;   /* Transaction index */
+   /* Specific field and its data follows immediately after this struct in memory */
+};
+```
+
+**Field Descriptions:**
+
+This structure is used for in-memory message buffering before writing to the capture file. The `sig` field maps to specific message types via the `SOLCAP_SIG_MAP` macro.
+
+---
+
+## Message Payload Structures
+
+### Account Update Header
+
+```c
+struct fd_solcap_account_update_hdr {
+   fd_pubkey_t              key;      /* Account public key (32 bytes) */
+   fd_solana_account_meta_t info;     /* Account metadata */
+   uint64_t                 data_sz;  /* Account data size in bytes */
+};
+
+struct __attribute__((packed)) fd_solana_account_meta {
+  ulong lamports;
+  uchar owner[32];
+  uchar executable;
+  uchar padding[3];
+};
+```
+
+**Field Descriptions:**
+
+- `key`: The 32-byte public key of the account being updated
+- `info`: Account metadata including lamports, owner, executable flag, rent epoch
+- `data_sz`: Size of the account data in bytes
+
+**Note**: The account data immediately follows this structure in the same packet. The data is `data_sz` bytes long.
+
+**Message Encoding**: Account updates are encoded as two consecutive messages:
+1. `SOLCAP_WRITE_ACCOUNT_HDR` with this header structure + account data
+2. Followed by any necessary padding to 4-byte boundary
+
+### Bank Preimage
+
+```c
+struct fd_solcap_bank_preimage {
+   fd_hash_t bank_hash;                    /* Bank hash */
+   fd_hash_t prev_bank_hash;               /* Previous bank hash */
+   fd_hash_t accounts_lt_hash_checksum;    /* Accounts hash */
+   fd_hash_t poh_hash;                     /* Proof of History hash */
+   uint64_t  signature_cnt;                /* Number of signatures */
+};
+```
+
+**Field Descriptions:**
+
+- `bank_hash`: The hash of the bank state after processing this slot
+- `prev_bank_hash`: The hash of the parent bank (previous slot)
+- `accounts_lt_hash_checksum`: Checksum of the accounts hash (Merkle root of account states)
+- `poh_hash`: The Proof of History hash for this slot
+- `signature_cnt`: Total number of signatures processed in this slot
+
+**Purpose**: This captures the complete state transition of a bank, enabling verification that execution was deterministic and produced the correct hash.
+
+## Alignment and Padding
+
+### Block Alignment
+
+All PCapNG blocks must be aligned to **4-byte boundaries**. This includes:
+- Section Header Block (28 bytes, naturally aligned)
+- Interface Description Block (20 bytes, naturally aligned)
+- Enhanced Packet Blocks (variable length, must be padded)
```

Comment:
Redundant, this is already specified in pcapng. IDB can have option so it's variable length. I would just remove this section